File: src/CazaTalentos/zbasket_02.py
import  numpy as np

np.random.seed(102191)

# calcula cuantos encestes logra un jugadora con indice de enceste prob
# haciendo qyt tiros libres

def ftirar(prob, qty):
  return sum(np.random.rand(qty) < prob)



# defino los jugadoras
mejor = 0.7
peloton = np.array(range(501, 600)) / 1000
jugadoras = np.append(mejor, peloton)

# veo que tiene el vector
jugadoras

# vectorizo la funcion  ftirar
vec_ftirar = np.vectorize(ftirar)

# hago que las 100 jugadoras tiren 10 veces cada una
vec_ftirar(jugadoras, 10)

primera_ganadora = 0

for i in range(10000): # diez mil experimentos
  vaciertos = vec_ftirar(jugadoras, 10) # 10 tiros libres cada jugadora
  mejor_ronda = np.argmax(vaciertos)
  if mejor_ronda == 0:
    primera_ganadora += 1




print(primera_ganadora)

File: src/CazaTalentos/zbasket_02.r


set.seed(102191)

# calcula cuantos encestes logra una jugadora con indice de enceste prob
# haciendo qyt tiros libres

ftirar <- function(prob, qty) {
  return(sum(runif(qty) < prob))
}


# defino los jugadores
mejor <- 0.7
peloton <- (501:599) / 1000
jugadoras <- c(mejor, peloton)

# veo que tiene el vector
jugadoras




# hago que los 100 jugadores tiren 10 veces cada uno
mapply(ftirar, jugadoras, 10)

primera_ganadora <- 0

for (i in 1:10000) { # diez mil experimentos

  vaciertos <- mapply(ftirar, jugadoras, 10) # 10 tiros libres cada jugador

  mejor_ronda <- which.max(vaciertos)
  if (mejor_ronda == 1) primera_ganadora <- primera_ganadora + 1
}


print(primera_ganadora)
